lib/Service/SocialApiService.php

```diff
@@ -182,4 +215,82 @@ public function updateContact(string $addressbookId, string $contactId, string $
 		}
 		return new JSONResponse([], Http::STATUS_OK);
 	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Updates social profile data for all contacts of an addressbook
+	 * // TODO: how to exclude certain contacts?
+	 *
+	 * @param {String} network the social network to use (fallback: take first match)
+	 * @param {String} user the address book owner
+	 *
+	 * @returns {JSONResponse} JSONResponse with the list of changed and failed contacts
+	 */
+	public function updateAddressbooks(string $network, string $user) : JSONResponse {
+
+		// double check!
+		$isAdminAllowed = $this->config->getAppValue(Application::APP_ID, 'allowSocialSync', 'yes');
+		if (!($isAdminAllowed === 'yes')) {
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+		$isUserEnabled = $this->config->getUserValue($user, Application::APP_ID, 'enableSocialSync', 'no');
+		if ($isUserEnabled !== 'yes') {
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+
+		$delay = 1;
+		$response = [
+			'updated' => [],
+			'checked' => [],
+			'failed' => [],
+		];
+
+		// get corresponding addressbook
+		$this->registerAddressbooks($user, $this->manager);
+
+		$addressBooks = $this->manager->getUserAddressBooks();
+		// TODO: filter out system addr books
+		// TODO: filter out deactivated books
+		// TODO: keep only owned address books (not shared ones)
+
+		foreach ($addressBooks as $addressBook) {
+			if (is_null($addressBook)) {
+				break;
+			}
+
+			// get contacts in that addressbook
+			$contacts = $addressBook->search('', ['UID'], ['types' => true]);
+			// TODO: filter for contacts with social profiles
```

Comment:
Could be optimized with `$addressBook->search('', ['X-SOCIALPROFILE'], ['types' => true]);`, if `X-SOCIALPROFILE` was [indexed](https://github.com/nextcloud/server/blob/master/apps/dav/lib/CardDAV/CardDavBackend.php#L78-L80)

Reply:
> Could be optimized with `$addressBook->search('', ['X-SOCIALPROFILE'], ['types' => true]);`, if `X-SOCIALPROFILE` was [indexed](https://github.com/nextcloud/server/blob/master/apps/dav/lib/CardDAV/CardDavBackend.php?rgh-link-date=2020-07-31T19%3A06%3A55Z#L78-L80)

Mind to add this? :)

Reply:
if it's not indexed (for older version) does it still works? Is the searched value just ignored?
If so it's ok I guess :)

Reply:
Right, I have included this and added a fallback for older server versions. But concerning the indexer, I could not find out how to add the `X-SOCIALPROFILE`. Simply adding it to the linked array did not have any effect. Do I need to trigger the indexer? Maybe someone from the server-team can help?

Reply:
Pull request created: https://github.com/nextcloud/server/pull/22085

Reply:
Turns out it was not cron, but me refreshing the profiles and triggering a re-index by that. We'll need a background job to re-index when a user activates this function to make sure all contacts are found. @skjnldsv or is there a better way / place to do that?